### ISSUE TITLE
 Incorrect compilation of Object Literal with Special Characters

### ISSUE THREAD

**Issue by t-k** [swc-project/swc#8940]
### Describe the bug

Since version 1.4.12, when constructing an object containing keys with special characters such as "`・`" ([U+30FB KATAKANA MIDDLE DOT](https://www.compart.com/en/unicode/U+30FB)), it will be converted to a format with no quotes.

This leads to a "Uncaught SyntaxError: Invalid or unexpected token" error during execution.

### Input code

```typescript
console.log({
  "あ": 1,
  "あ・あ": 1,
  "・あ": 1,
  "・": 1,
  "あ・": 1,
})
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false
    },
    "target": "es5",
    "loose": false,
    "minify": {
      "compress": {
        "arguments": false,
        "arrows": true,
        "booleans": true,
        "booleans_as_integers": false,
        "collapse_vars": true,
        "comparisons": true,
        "computed_props": true,
        "conditionals": true,
        "dead_code": true,
        "directives": true,
        "drop_console": false,
        "drop_debugger": true,
        "evaluate": true,
        "expression": false,
        "hoist_funs": false,
        "hoist_props": true,
        "hoist_vars": false,
        "if_return": true,
        "join_vars": true,
        "keep_classnames": false,
        "keep_fargs": true,
        "keep_fnames": false,
        "keep_infinity": false,
        "loops": true,
        "negate_iife": true,
        "properties": true,
        "reduce_funcs": false,
        "reduce_vars": false,
        "side_effects": true,
        "switches": true,
        "typeofs": true,
        "unsafe": false,
        "unsafe_arrows": false,
        "unsafe_comps": false,
        "unsafe_Function": false,
        "unsafe_math": false,
        "unsafe_symbols": false,
        "unsafe_methods": false,
        "unsafe_proto": false,
        "unsafe_regexp": false,
        "unsafe_undefined": false,
        "unused": true,
        "const_to_let": true,
        "pristine_globals": true
      },
      "mangle": false
    }
  },
  "module": {
    "type": "es6"
  },
  "minify": false,
  "isModule": true
}
```

### Playground link (or link to the minimal reproduction)

https://play.swc.rs/?version=1.4.12&code=H4sIAAAAAAAAA0vOzyvOz0nVy8lP16jmUlBQetzYpGSlYKgDZT9u3o0igsFFUwzh1moCABZnot9YAAAA&config=H4sIAAAAAAAAA32UO5LbMAxA%2Bz2FR3XapMgBtssZODQJytxQhIYAvdbs%2BO4BJfmTNaROwsOP%2BH29HQ7dB7nu9%2BFLPuVntIWg3P9FQlNmexFJx9MI5Eocuftxo0wNBZsIZtF1IR3b0gM3K6Cfq3qXEAlu6qtsiDmG6Tmgw2EsQPQkE6n4qwNkpv%2FtV1bwswEu9Vl%2BRExg8w4xlkzMDD0UzbHDlOxIYM62KF5aprZEQi1Eg5XBm7HgqPLsI0fMEvOVerDeOPSgoFjAcTyDZiaxxCyTPE95z4w9HGvfz03%2BZg1nm6plJSZc5pZItorXE0ZiE2rWSrjAjRoscC3ud8sYTAGuJb%2FafWDMGz35CyAVSJYo2wE0v7NGkHnasg67ljEHGVmeFC7zrb0yQy9FNTEGpbKtMlA4at0s4KuDVlmnpbPijfJR9GAgBJkVxTV9RnYnLWjbcgwKkP7aoE3VAsx9Czd4W4gd%2FC6vZH3AVo3B8mmb0jQcMe0EGIBP6HcUpBWM27jIlbiM27xmDzIa4FWVSjN4PQKyAIwmzcfyZTZkPcSj6RMeH2diVbjej%2FBgc%2F%2FY9%2BUOv60K3YC%2BznC98K2%2Fy13%2B1T2Ublf4nngX6c%2FNcg56%2FQckCXpSLQYAAA%3D%3D

### Expected behavior

```js
console.log({
    あ: 1,
    "あ・あ": 1,
    "・あ": 1,
    "・": 1,
    "あ・": 1
});

```

### Actual behavior

```js
console.log({
    あ: 1,
    あ・あ: 1,
    "・あ": 1,
    "・": 1,
    あ・: 1
});
```

### Version

Version 1.4.12 and above

**Comment by kdy1** [comment]
cc @Boshen It seems like https://github.com/swc-project/swc/pull/8785 introduced a regression. Any ideas about the difference?

**Comment by Boshen** [comment]
@kdy1 Can you point me to where you're dropping the quotes? I want to check whether it's using the correct functions or not.

**Comment by kdy1** [comment]
@Boshen 

https://github.com/swc-project/swc/blob/6d7b99b4824db3834e3aa93ab50951f3c2eefd83/crates/swc_ecma_minifier/src/compress/pure/properties.rs#L84-L96

**Comment by Boshen** [comment]
I found part of the problem:

U+30FB was added to `ID_Continue` from Unicode version 14 to 15, but v8 is using an older version of unicode ... which I can't find the exact version.

I spent 2 hours searching for all the missing pieces, I give up.

I'll publish a version with unicode 14.

**Comment by Boshen** [comment]
@kdy1 Can you change and pin the version (perhaps with the comment as well so we are clear which version of unicode we are targeting.)

```
unicode-id-start = "=1.0.4" # Unicode 14.0.0
```

**Comment by t-k** [comment]
@kdy1 @Boshen I have confirmed the fix in v1.5.6. Thank you for your quick response!
